services/mailer/mail.go

```diff
@@ -75,6 +75,12 @@ func sendUserMail(language string, u *user_model.User, tpl base.TplName, code, s
 		"Language":          locale.Language(),
 	}
 
+	subjectFromTemplate := subjectFromTemplate(string(tpl), data)
+	if subjectFromTemplate != "" {
+		subject = subjectFromTemplate
+	}
+	data["Subject"] = subject
```

Comment:
Can't we extract this into a new function `setSubject` that takes as parameters `data, defaultSubject` and always call that instead?

Reply:
Make sense. Will change it.

Reply:
Em...Now I think the current way is better. 
`subject` var is also used in `NewMessage(u.Email, subject, content.String())`. If using `setSubject` and removing the `subject` var, then we should use `NewMessage(u.Email, data["Subject"].(string), content.String())` instead.  I don't know if it's a good idea to use type assertion in such simple code.